Client/SiteImageHelper.swift

```diff
@@ -18,8 +18,16 @@ enum SiteImageType: Int {
     }
 }
 
+protocol SiteImageHelperProtocol {
+    func fetchImageFor(site: Site,
+                       imageType: SiteImageType,
+                       shouldFallback: Bool,
+                       metadataProvider: LPMetadataProvider,
```

Comment:
wondering if it would be worth it to have a default param for `metadataProvider: LPMetadataProvider = LPMetadataProvider()` in a protocol extension? Since we would only replace this in tests?

Reply:
I can't picture what that would look like. Can you provide a code example of what the extension would look like?

Reply:
I made a little playground 😄 
```
import Foundation

protocol Fruit {
    func takeABite(ofSize size: Int)
}

extension Fruit {
    func takeABite(ofSize size: Int = 10) {
        self.takeABite(ofSize: size)
    }
}

class Apple: Fruit {
    // We can use default param of 10 with the protocol
    // or provide a custom bite size
    func takeABite(ofSize size: Int) {
        print("Taking a bite of size \(size)")
    }
}

class Orange: Fruit {
    // Will ignore default param of 20 here
    // unless we have type Orange directly.
    func takeABite(ofSize size: Int = 20) {
        print("Taking a bite of size \(size)")
    }
}

let fruits: [Fruit] = [Apple(), Orange()]
for fruit in fruits {
    fruit.takeABite()
}

```

Reply:
🤯 I had no idea you could this this!